ts/src/renderer/phaser/classes/PhaserRegion.ts

```diff
@@ -1,62 +1,64 @@
 class PhaserRegion extends Phaser.GameObjects.Graphics {
 
-	private width: number;
-	private height: number;
+	private updateDimensionsListener: EvtListener;
+	private destroyListener: EvtListener;
 
 	constructor (
 		scene: Phaser.Scene,
 		private region: Region
 	) {
 		super(scene);
 
-		const stats = this.region._stats.default
+		const stats = this.region._stats.default;
 
-		// draw rectangle
-		const width = this.width = stats.width;
-		const height = this.height = stats.height;
-		this.fillStyle(0xFF0000, 0.4);
+		// Phaser wants a number for these
+		this.fillStyle(
+			Number(`0x${stats.inside.substring(1)}`),
+			stats.alpha / 100 || 0.4
+		);
 		this.fillRect(
 			0,
 			0,
-			width,
-			height
+			stats.width,
+			stats.height
 		);
 
 		this.x = stats.x;
 		this.y = stats.y;
 
 		scene.add.existing(this);
 		scene.events.on('update', this.update, this);
-	}
-
-	update (/*time: number, delta: number*/): void {
 
-		const region = this.region;
-		const container = region.regionUi._pixiContainer;
+		this.updateDimensionsListener = region.on('update-region-dimensions', () => {
 
-		if (region._destroyed || container._destroyed) {
-			this.scene.events.off('update', this.update, this);
-			this.destroy();
-			return;
-		}
-
-		const stats = this.region._stats.default
+			const stats = this.region._stats.default;
```

Comment:
lines 12-27 and 34-51 are basically duplicates. can you move this logic into a class method and then invoke it once in constructor and on each size update event?

Reply:
heard and agree 👍